File: DragonzLand/DragonzLand.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace DragonzLandBot
{

    public class DragonzLandQuery
    {
        public int Index { get; set; }
        public string Name { get; set; }
        public string Auth { get; set; }
    }

    public class DragonzLandAuthRequest
    {
        [JsonPropertyName("initData")]
        public string InitData { get; set; }
    }

    public class DragonzLandAuthResponse
    {
        [JsonPropertyName("accessToken")]
        public string AccessToken { get; set; }
        [JsonPropertyName("refreshToken")]
        public string RefreshToken { get; set; }
    }

    public class DragonzLandMeResponse
    {
        [JsonPropertyName("id")]
        public string Id { get; set; }
        [JsonPropertyName("telegramUserId")]
        public long TelegramUserId { get; set; }
        [JsonPropertyName("username")]
        public string Username { get; set; }
        [JsonPropertyName("firstName")]
        public string FirstName { get; set; }
        [JsonPropertyName("lastName")]
        public string LastName { get; set; }
        [JsonPropertyName("diamonds")]
        public int Diamonds { get; set; }
        [JsonPropertyName("coins")]
        public int Coins { get; set; }
        [JsonPropertyName("power")]
        public int Power { get; set; }
        [JsonPropertyName("feedCoins")]
        public int FeedCoins { get; set; }
        [JsonPropertyName("level")]
        public int Level { get; set; }
        [JsonPropertyName("energy")]
        public int Energy { get; set; }
        [JsonPropertyName("energyLimit")]
        public int EnergyLimit { get; set; }
        [JsonPropertyName("tasks")]
        public List<DragonzLandMeTask> Tasks { get; set; }
        [JsonPropertyName("boosts")]
        public List<DragonzLandMeBoost> Boosts { get; set; }
        [JsonPropertyName("cards")]
        public List<DragonzLandMeCard> Cards { get; set; }
    }

    public class DragonzLandMeTask
    {
        [JsonPropertyName("taskId")]
        public string TaskId { get; set; }
        [JsonPropertyName("level")]
        public int Level { get; set; }
        [JsonPropertyName("attempts")]
        public int Attempts { get; set; }
        [JsonPropertyName("attemptedAt")]
        public DateTime? AttemptedAt { get; set; }
        [JsonPropertyName("activeAt")]
        public DateTime? ActiveAt { get; set; }
    }

    public class DragonzLandMeBoost
    {
        [JsonPropertyName("boostId")]
        public string BoostId { get; set; }
        [JsonPropertyName("level")]
        public int Level { get; set; }
        [JsonPropertyName("attempts")]
        public int Attempts { get; set; }
        [JsonPropertyName("attemptedAt")]
        public DateTime? AttemptedAt { get; set; }
        [JsonPropertyName("activeAt")]
        public DateTime? ActiveAt { get; set; }
    }

    public class DragonzLandMeCard
    {
        [JsonPropertyName("id")]
        public string Id { get; set; }
        [JsonPropertyName("cardId")]
        public string CardId { get; set; }
        [JsonPropertyName("level")]
        public int Level { get; set; }
        [JsonPropertyName("attemptedAt")]
        public DateTime? AttemptedAt { get; set; }
        [JsonPropertyName("activeAt")]
        public DateTime? ActiveAt { get; set; }
    }

    public class DragonzLandTaskResponse
    {
        [JsonPropertyName("id")]
        public string Id { get; set; }
        [JsonPropertyName("type")]
        public string Type { get; set; }
        [JsonPropertyName("recurrence")]
        public string Recurrence { get; set; }
        [JsonPropertyName("attemptsLimit")]
        public int AttemptsLimit { get; set; }
        [JsonPropertyName("active")]
        public bool Active { get; set; }
        [JsonPropertyName("featured")]
        public bool Featured { get; set; }
        [JsonPropertyName("pinned")]
        public bool Pinned { get; set; }
        [JsonPropertyName("listed")]
        public bool Listed { get; set; }
        [JsonPropertyName("title")]
        public string Title { get; set; }
    }

    public class DragonzLandTaskDoneRequest
    {
        [JsonPropertyName("taskId")]
        public string TaskId { get; set; }
    }

    public class DragonzLandFeedRequest
    {
        [JsonPropertyName("feedCount")]
        public int FeedCount { get; set; }
    }

    public class DragonzLandBoostResponse
    {
        [JsonPropertyName("id")]
        public string Id { get; set; }
        [JsonPropertyName("type")]
        public string Type { get; set; }
        [JsonPropertyName("active")]
        public bool Active { get; set; }
        [JsonPropertyName("comingSoon")]
        public bool ComingSoon { get; set; }
        [JsonPropertyName("title")]
        public string Title { get; set; }
        [JsonPropertyName("attemptsLimit")]
        public int AttemptsLimit { get; set; }
        [JsonPropertyName("recurrence")]
        public string Recurrence { get; set; }
    }

    public class DragonzLandBoostBuyRequest
    {
        [JsonPropertyName("boostId")]
        public string BoostId { get; set; }
    }

    public class DragonzLandCardResponse
    {

    }
}

File: DragonzLand/DragonzLand.vb
Imports System.Text.Json.Serialization

Public Class DragonzLandQuery
    Public Property Index As Integer
    Public Property Name As String
    Public Property Auth As String
End Class

Public Class DragonzLandAuthRequest
    <JsonPropertyName("initData")>
    Public Property InitData As String
End Class

Public Class DragonzLandAuthResponse
    <JsonPropertyName("accessToken")>
    Public Property AccessToken As String
    <JsonPropertyName("refreshToken")>
    Public Property RefreshToken As String
End Class

Public Class DragonzLandMeResponse
    <JsonPropertyName("id")>
    Public Property Id As String
    <JsonPropertyName("telegramUserId")>
    Public Property TelegramUserId As Long
    <JsonPropertyName("username")>
    Public Property Username As String
    <JsonPropertyName("firstName")>
    Public Property FirstName As String
    <JsonPropertyName("lastName")>
    Public Property LastName As String
    <JsonPropertyName("diamonds")>
    Public Property Diamonds As Integer
    <JsonPropertyName("coins")>
    Public Property Coins As Integer
    <JsonPropertyName("power")>
    Public Property Power As Integer
    <JsonPropertyName("feedCoins")>
    Public Property FeedCoins As Integer
    <JsonPropertyName("level")>
    Public Property Level As Integer
    <JsonPropertyName("energy")>
    Public Property Energy As Integer
    <JsonPropertyName("energyLimit")>
    Public Property EnergyLimit As Integer
    <JsonPropertyName("tasks")>
    Public Property Tasks As List(Of DragonzLandMeTask)
    <JsonPropertyName("boosts")>
    Public Property Boosts As List(Of DragonzLandMeBoost)
    <JsonPropertyName("cards")>
    Public Property Cards As List(Of DragonzLandMeCard)
End Class

Public Class DragonzLandMeTask
    <JsonPropertyName("taskId")>
    Public Property TaskId As String
    <JsonPropertyName("level")>
    Public Property Level As Integer
    <JsonPropertyName("attempts")>
    Public Property Attempts As Integer
    <JsonPropertyName("attemptedAt")>
    Public Property AttemptedAt As DateTime?
    <JsonPropertyName("activeAt")>
    Public Property ActiveAt As DateTime?
End Class

Public Class DragonzLandMeBoost
    <JsonPropertyName("boostId")>
    Public Property BoostId As String
    <JsonPropertyName("level")>
    Public Property Level As Integer
    <JsonPropertyName("attempts")>
    Public Property Attempts As Integer
    <JsonPropertyName("attemptedAt")>
    Public Property AttemptedAt As DateTime?
    <JsonPropertyName("activeAt")>
    Public Property ActiveAt As DateTime?
End Class

Public Class DragonzLandMeCard
    <JsonPropertyName("id")>
    Public Property Id As String
    <JsonPropertyName("cardId")>
    Public Property CardId As String
    <JsonPropertyName("level")>
    Public Property Level As Integer
    <JsonPropertyName("attemptedAt")>
    Public Property AttemptedAt As DateTime?
    <JsonPropertyName("activeAt")>
    Public Property ActiveAt As DateTime?
End Class

Public Class DragonzLandTaskResponse
    <JsonPropertyName("id")>
    Public Property Id As String
    <JsonPropertyName("type")>
    Public Property Type As String
    <JsonPropertyName("recurrence")>
    Public Property Recurrence As String
    <JsonPropertyName("attemptsLimit")>
    Public Property AttemptsLimit As Integer
    <JsonPropertyName("active")>
    Public Property Active As Boolean
    <JsonPropertyName("featured")>
    Public Property Featured As Boolean
    <JsonPropertyName("pinned")>
    Public Property Pinned As Boolean
    <JsonPropertyName("listed")>
    Public Property Listed As Boolean
    <JsonPropertyName("title")>
    Public Property Title As String
End Class

Public Class DragonzLandTaskDoneRequest
    <JsonPropertyName("taskId")>
    Public Property TaskId As String
End Class

Public Class DragonzLandFeedRequest
    <JsonPropertyName("feedCount")>
    Public Property FeedCount As Integer
End Class

Public Class DragonzLandBoostResponse
    <JsonPropertyName("id")>
    Public Property Id As String
    <JsonPropertyName("type")>
    Public Property Type As String
    <JsonPropertyName("active")>
    Public Property Active As Boolean
    <JsonPropertyName("comingSoon")>
    Public Property ComingSoon As Boolean
    <JsonPropertyName("title")>
    Public Property Title As String
    <JsonPropertyName("attemptsLimit")>
    Public Property AttemptsLimit As Integer
    <JsonPropertyName("recurrence")>
    Public Property Recurrence As String
End Class

Public Class DragonzLandBoostBuyRequest
    <JsonPropertyName("boostId")>
    Public Property BoostId As String
End Class

Public Class DragonzLandCardResponse

End Class